components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApplicationsApiServiceImpl.java

```diff
@@ -50,55 +46,35 @@ public Response applicationsGet(String subscriber, String groupId, String limit,
 
     @Override
     public Response applicationsPost(ApplicationDTO body, String contentType) {
-        //todo: validation, need to be moved
-        String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
-        if (StringUtils.isEmpty(body.getName())) {
-            return Response.serverError().entity(new ApiResponseMessage(ApiResponseMessage.ERROR, "Name is empty"))
-                    .build();
-        }
-        if (StringUtils.isEmpty(body.getThrottlingTier())) {
-            return Response.serverError().entity(new ApiResponseMessage(ApiResponseMessage.ERROR, "Tier is empty"))
-                    .build();
-        }
-        String subscriber = body.getSubscriber(); //todo: implement permission
-        if (subscriber == null) { 
-            subscriber = username;
-        }
-
+        String username = RestApiUtil.getLoggedInUsername();
+        String subscriber = body.getSubscriber();
         try {
             APIConsumer apiConsumer = APIManagerFactory.getInstance().getAPIConsumer(username);
-            Application application = ApplicationMappingUtil.fromDTOtoApplication(body, new Subscriber(subscriber));
-            String status = apiConsumer.addApplication(application, subscriber);  //todo: use "status" properly
-            return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, status)).build();
-        } catch (APIManagementException e) {
+            Application application = ApplicationMappingUtil.fromDTOtoApplication(body);
+            int applicationId = apiConsumer.addApplication(application, subscriber);
+
+            //retrieves the created application and send as the response
+            Application createdApplication = apiConsumer.getApplicationById(applicationId);
+            ApplicationDTO createdApplicationDTO = ApplicationMappingUtil.fromApplicationtoDTO(createdApplication);
+
+            //to be set as the Location header
+            URI location = new URI(RestApiConstants.RESOURCE_PATH_APPLICATIONS + "/" +
+                    createdApplicationDTO.getApplicationId());
+            return Response.created(location).entity(createdApplicationDTO).build();
+        } catch (APIManagementException | URISyntaxException e) {
             throw new InternalServerErrorException(e);
```

Comment:
Fixed in commit https://github.com/wso2/carbon-apimgt/commit/10b904553097e950f881465d125ef962ac32a3a3